StringCalculator/StringCalculator.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace StringCalculator;
+public class StringCalculator
+{
+    private string[] seperators = new string[] { ",", "\\n" };
+    private List<string> negativeNumbers = new List<string>();
+    private int maxNumberToAdd = 1000;
+    private int finalResult = 0;
+
+    public int Add(string numbersGiven)
+    {
+        string numbersToCalculate = numbersGiven;
+
+        if (String.IsNullOrEmpty(numbersToCalculate))
+        {
+            return 0;
+        }
+
+        numbersToCalculate = CheckForSeperators(numbersToCalculate);
+        String[] splitNumbersToAdd = numbersToCalculate.Split(seperators, StringSplitOptions.None);
+
+        CalculateSplitNumberAddition(splitNumbersToAdd);
+
+        return finalResult;
+    }
+
+    private void CalculateSplitNumberAddition(string[] splitNumbersToAdd)
+    {
+        foreach (string currentNumber in splitNumbersToAdd)
+        {
+            int convertedCurrentNumber = Convert.ToInt32(currentNumber);
+
+            if (convertedCurrentNumber < 0)
+            {
+                negativeNumbers.Add(currentNumber);
+                continue;
+            }
+            else if (convertedCurrentNumber > maxNumberToAdd)
+            {
+                continue;
+            }
+
+            finalResult += convertedCurrentNumber;
+        }
+
+        if (negativeNumbers.Count > 0)
+        {
+            string message = "Negatives not allowed: " + String.Join(",", negativeNumbers);
+            throw new Exception(message);
+        }
+    }
+
+    private string CheckForSeperators(string numbersToCalculate)
```

Comment:
Generally, if you method returns a value you want to make that clear from it's name - "Check" doesn't suggest that a value is going to be returned.